src/core.ts

```diff
@@ -0,0 +1,370 @@
+import YAML from "yaml"
+
+import { commitStateFailure, commitStateSuccess } from "./constants"
+import { LoggerInterface } from "./logger"
+import { Octokit, PR, RuleUserInfo } from "./types"
+import { configurationSchema } from "./validation"
+
+const combineUsers = async function (
+  pr: PR,
+  client: Octokit,
+  presetUsers: string[],
+  teams: string[],
+) {
+  const users: Map<string, RuleUserInfo> = new Map()
+
+  for (const user of presetUsers) {
+    if (pr.user.login != user) {
+      users.set(user, { team: null })
+    }
+  }
+
+  for (const team of teams) {
+    const teamMembersResponse = await client.rest.teams.listMembersInOrg({
+      org: pr.base.repo.owner.login,
+      team_slug: team,
+    })
+    if (teamMembersResponse.status !== 200) {
+      throw new Error(`Failed to fetch team members from ${team}`)
+    }
+
+    for (const member of teamMembersResponse.data) {
+      if (member === null) {
+        continue
+      }
+      if (
+        pr.user.login != member.login &&
+        // We do not want to register a team for this user if their approval is
+        // supposed to be requested individually
+        users.get(member.login) === undefined
+      ) {
+        users.set(member.login, { team })
+      }
+    }
+  }
+
+  return users
+}
+
+// This function should only depend on its inputs so that it can be tested
+// without inconveniences. If you need more external input then pass it as a
+// function argument.
+export const runChecks = async function (
```

Comment:
I think 300 lines is too much for a single function. Is there a way to break it down? I'm having a hard time understanding what's going on in there

Reply:
I don't have a problem with the function as it is but added the feedback as a suggestion in #44. I don't want to make changes here because the following PRs build on this.